sections/main-product.liquid

```diff
@@ -106,6 +106,51 @@
+                {{ block.shopify_attributes }}
+                id="Inventory-{{ section.id }}"
+                role="status"
+              >
+                {%- if product.selected_or_first_available_variant.inventory_quantity != 0 and product.selected_or_first_available_variant.inventory_management == 'shopify' -%}
+                  {%- if product.selected_or_first_available_variant.inventory_quantity <= block.settings.inventory_threshold -%}
+                    <svg width="15" height="15" role="presentation">
+                      <circle cx="7.5" cy="7.5" r="7.5" fill="rgb(238,148,65, 0.3)"/>
+                      <circle cx="7.5" cy="7.5" r="5" stroke="rgb(255, 255, 255)" stroke-width="1" fill="rgb(238,148,65)"/>
+                    </svg>
+                    {%- if block.settings.show_inventory_quantity -%}
+                      {{- 'products.product.inventory_low_stock_show_count' | t: quantity: product.selected_or_first_available_variant.inventory_quantity -}}
+                    {%- else -%}
+                      {{- 'products.product.inventory_low_stock' | t -}}
+                    {%- endif -%}
+                  {%- else -%}
+                    <svg width="15" height="15" role="presentation">
+                      <circle cx="7.5" cy="7.5" r="7.5" fill="rgb(62,214,96, 0.3)"/>
+                      <circle cx="7.5" cy="7.5" r="5" stroke="rgb(255, 255, 255)" stroke-width="1" fill="rgb(62,214,96)"/>
+                    </svg>
+                    {%- if block.settings.show_inventory_quantity -%}
+                      {{- 'products.product.inventory_in_stock_show_count' | t: quantity: product.selected_or_first_available_variant.inventory_quantity -}}
+                    {%- else -%}
+                        {{- 'products.product.inventory_in_stock' | t -}}
+                    {%- endif -%}
+                  {%- endif -%}
+                {%- elsif  product.selected_or_first_available_variant.inventory_management == 'shopify' -%}
+                  {%- if product.selected_or_first_available_variant.inventory_policy == 'continue' -%}
+                    <svg width="15" height="15" role="presentation">
+                      <circle cx="7.5" cy="7.5" r="7.5" fill="rgb(62,214,96, 0.3)"/>
+                      <circle cx="7.5" cy="7.5" r="5" stroke="rgb(255, 255, 255)" stroke-width="1" fill="rgb(62,214,96)"/>
+                    </svg>
+                    {{- 'products.product.inventory_in_stock' | t -}}
+                  {%- else -%}
+                    <svg width="15" height="15" role="presentation">
+                      <circle cx="7.5" cy="7.5" r="7.5" fill="rgb(200,200,200, 0.3)"/>
+                      <circle cx="7.5" cy="7.5" r="5" stroke="rgb(255, 255, 255)" stroke-width="1" fill="rgb(200,200,200)"/>
+                    </svg>
+                    {{- 'products.product.inventory_out_of_stock' | t -}}
+                  {%- endif -%}
+                {%- endif -%}
+              </p>
```

Comment:
I feel like this code block could be cleaned up to reduce it’s cognitive complexity. First of all it seems this should only render anything when `product.selected_or_first_available_variant.inventory_management == 'shopify'` so that could just wrap the whole block to simplify the inner conditions. Also, it might be better to have the base case come first (`inventory_quantity == 0`)
Secondly, it might clean things up a bit to have a liquid block at the top of this and assign some of the values and booleans to variables to use inside here.

I personally feel like these changes would make this block way more readable, and therefore maintainable

Reply:
Yeah there is quite a few checks. Not sure it was worth the assigns. But maybe it would make everything easier to read.